arangod/Cluster/Maintenance.cpp

```diff
@@ -611,21 +649,38 @@ arangodb::Result arangodb::maintenance::executePlan(VPackSlice const& plan,
     // open ACTIONS
     TRI_ASSERT(report.isOpenObject());
     report.add(ACTIONS, VPackValue(VPackValueType::Array));
-  
-    // enact all
-    for (auto& action : actions) {
-      LOG_TOPIC("8513c", DEBUG, Logger::MAINTENANCE)
-          << "adding action " << action.get() << " to feature ";
+  }
+
+  // enact all
+  for (auto& action : actions) {
+    LOG_TOPIC("8513c", DEBUG, Logger::MAINTENANCE)
+        << "adding action " << action.get() << " to feature ";
+    if (debugActions) {
       VPackObjectBuilder b(&report);
       action->toVelocyPack(report);
+    }
+    if (!action->isForced()) {
       feature.addAction(std::move(action), false);
+    } else {
+      std::string shardName = action->get(SHARD);
+      bool ok = feature.lockShard(shardName, action);
+      TRI_ASSERT(ok);
+      try {
+        Result res = feature.addAction(std::move(action), false);
+        if (res.fail()) {
+          feature.unlockShard(shardName);
+        }
+      } catch (std::exception const& exc) {
+        feature.unlockShard(shardName);
+        LOG_TOPIC("86762", INFO, Logger::MAINTENANCE)
```

Comment:
Should the exception be rethrown from here, or should we just go on normally?
If all actions are independent, we probably can go on here...